api/genericresource/parse.go

```diff
@@ -12,35 +13,82 @@ func newParseError(format string, args ...interface{}) error {
 	return fmt.Errorf("could not parse GenericResource: "+format, args...)
 }
 
-// Parse parses the GenericResource resources given by the arguments
-func Parse(cmd string) ([]*api.GenericResource, error) {
-	var rs []*api.GenericResource
+func namedResourceVal(res string) (int64, error) {
+	return strconv.ParseInt(res, 10, 64)
+}
+
+func allNamedResources(res []string) bool {
+	for _, v := range res {
+		if _, err := namedResourceVal(v); err == nil {
+			return false
+		}
+	}
+
+	return true
+}
+
+// ParseCmd parses the Generic Resource command line argument
+// and returns a list of *api.GenericResource
+func ParseCmd(cmd string) ([]*api.GenericResource, error) {
+	if strings.Contains(cmd, "\n") {
```

Comment:
Seems like a strange thing to check for - I don't know of any CLI arguments where this would be an issue, or where we explicitly check for it. But it shouldn't hurt.